### PR TITLE
Plugins: Use `wp_get_wp_version()` in `is_wp_version_compatible()`.

### PR DESCRIPTION
**Pull request by costdev**
This implements the new `wp_get_wp_version()` helper function to get the unmodified WordPress version for compatibility checks.

A new global, `_wp_tests_wp_version` has been introduced to allow for testing against different "current" WordPress versions.

Trac ticket: https://core.trac.wordpress.org/ticket/61781

### CODE REVIEW

File: src/wp-includes/functions.php [hunk 1]
```diff
@@ -8818,7 +8818,15 @@ function clean_dirsize_cache( $path ) {
  * @return bool True if required version is compatible or empty, false if not.
  */
 function is_wp_version_compatible( $required ) {
-	global $wp_version;
```
Comment (mukeshpanchal27) [inline src/wp-includes/functions.php:8842]: Remove `@global` docblock for `$wp_version` and add docblock for `_wp_tests_wp_version`

Comment (costdev) [reply src/wp-includes/functions.php:8842]: Done in [857128f](https://github.com/WordPress/wordpress-develop/pull/7104/commits/857128fc26530f2b1405518b6b07ba976ea90c15).

**Review by peterwilsoncc — APPROVED** [review]
This looks good to me, thanks Colin.

**Review by mukeshpanchal27 — APPROVED** [review]
The change looks good to me.

**Review by SergeyBiryukov — APPROVED** [review]

**Comment by peterwilsoncc** [comment]
Merged [r58843](https://core.trac.wordpress.org/changeset/58843) / https://github.com/WordPress/wordpress-develop/commit/0e06e2b522e812e0d62193eb7ad21f9b38dde7bb

**Pull request closed without merging**